Main.py

```diff
@@ -541,17 +543,25 @@ def create_playthrough(spoiler):
     # Regenerate the spheres as we might not reach places the same way anymore.
     playthrough.reset()  # playthrough state has no items, okay to reuse sphere 0 cache
     collection_spheres = []
+    entrance_spheres = []
+    previously_accessed_entrances = []
     while True:
         # Not collecting while the generator runs means we only get one sphere at a time
         # Otherwise, an item we collect could influence later item collection in the same sphere
         collected = list(playthrough.iter_reachable_locations(required_locations))
+        accessed_entrances = [entrance for world in worlds for entrance in world.get_shuffled_entrances() 
```

Comment:
I wanted to do something similar at first but I don't know a clean way to do that in Python. I don't think there is any built-in way to partition/split a list based on a predicate, is there?
I would like to do something like this but i can't find a decent way to do it in Python:
```python
accessed, remaining = partition(lambda entrance: can_reach(entrance), remaining)
```
I guess it can be done with 2 consecutive filters but I'm curious to know if there is a better way.

Reply:
Not really a one-line partitioner, but I would do
```py
accessed = set(filter(can_reach, remaining))
remaining -= accessed
```